### PR TITLE
printrun: 1.6.0 -> 2.0.0rc5

### PR DESCRIPTION
**Pull request by alexfmpe**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/printrun/default.nix [hunk 1]
```diff
@@ -1,17 +1,16 @@
-{ stdenv, python27Packages, fetchFromGitHub }:
+{ stdenv, python36Packages, fetchFromGitHub }:
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/printrun/default.nix:1]:
```suggestion
{ stdenv, python3Packages, fetchFromGitHub }:
```

File: pkgs/applications/misc/printrun/default.nix [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
 
-python27Packages.buildPythonApplication rec {
-  name = "printrun-20150310";
+python36Packages.buildPythonApplication rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/printrun/default.nix:3]:
```suggestion
python3Packages.buildPythonApplication rec {
```

File: pkgs/applications/misc/printrun/default.nix [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
-    rev = name;
-    sha256 = "09ijv8h4k5h15swg64s7igamvynawz7gdi7hiymzrzywdvr0zwsa";
+  src = python36Packages.fetchPypi {
+    inherit pname version;
+    sha256 = "0kgwjjq7b9p8prs4mjrydfjc758sm1p1l598hmgdxjxizj6j06qs";
   };
 
-  propagatedBuildInputs = with python27Packages; [
-    wxPython30 pyserial dbus-python psutil numpy pyopengl pyglet cython
+  propagatedBuildInputs = with python36Packages; [
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/printrun/default.nix:12]:
```suggestion
  propagatedBuildInputs = with python3Packages; [
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please do not merge master into PRs.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 90272` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>printrun</li>
  </ul>
</details>

  Finished executing setuptoolsBuildPhase
  installing
  Executing pipInstallPhase
  /build/Printrun-2.0.0rc6/dist /build/Printrun-2.0.0rc6
  Processing ./Printrun-2.0.0rc6-cp36-cp36m-linux_x86_64.whl
  Requirement already satisfied: appdirs>=1.4.0 in /nix/store/svy5cp8xbbszaqc5hxm7hm4fkqcmj64z-python3.6-appdirs-1.4.4/lib/python3.6/site-packages (from Printrun==2.0.0rc6) (1.4.4)
  Requirement already satisfied: numpy>=1.8.2 in /nix/store/4j9h59l2716jvsfhrd0js51cp85q2pcx-python3.6-numpy-1.19.4/lib/python3.6/site-packages (from Printrun==2.0.0rc6) (1.19.4)
  Requirement already satisfied: psutil>=2.1 in /nix/store/mbfdg4clj0j5brjd7yjp0kz79913qfwy-python3.6-psutil-5.7.3/lib/python3.6/site-packages (from Printrun==2.0.0rc6) (5.7.3)
  ERROR: Could not find a version that satisfies the requirement cairocffi (from printrun)
  ERROR: No matching distribution found for cairocffi
cannot build derivation '/nix/store/wjyyvqxkhn3718sg2z2zgf0imdcxsp0z-env.drv': 1 dependencies couldn't be built

**Comment by alexfmpe** [comment]
2.0.0rc7 is now out but it has the same dbus-python error as 2.0.0rc6 so I'm pointing this PR back to 2.0.0rc5 which I managed to build.
Redid on top of current master and force pushed to remove intermediate broken state and the merge commit

**Comment by SuperSandro2000** [comment]
@alexfmpe please fix the merge conflict.

**Comment by alexfmpe** [comment]
@SuperSandro2000 Fixed

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 90272` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>printrun</li>
</ul>
</details>

**Pull request merged**